content/rust/concepts/strings/strings.md

```diff
@@ -0,0 +1,146 @@
+- 'Code Foundations'
+  CatalogContent:
+- 'learn-rust'
+- 'paths/computer-science'
+
+---
+
+**`Strings`** in Rust diverge a little bit from conventional programming languages. The approach to string manipulation
+in Rust is influenced by its systems-focused design. This concept entry will provide insights into creating and
+manipulating strings in Rust, offering practical examples to illustrate key aspects.
+
+Tackling data structures of variable size, such as strings, can get a bit tricky, and Rust has its spin on it. In Rust,
+a string is essentially a sequence of Unicode characters encoded in UTF-8. As an example - the string "Codecademy Rust
+concept," where each character is a valid Unicode entity – "C," "o," "d," "e," "c," and so on.
+
+## String types
+
+In Rust, there are two main types related to strings, each serving a specific purpose.
+
+1. **`String`**
+
+   - **Description:** A heap-allocated, growable, and mutable string type.
+   - **Usage:** Used when dynamic string manipulation and ownership transfer are required.
+   - **Example:**
+
+     ```rust
+     let mut dynamic_string = String::from("Hello, Rust!");
+     ```
+
+2. **`&str` (String Slice)**
+
+   - **Description:** A reference to a sequence of UTF-8 bytes, often used as a view into a `String` or a string
+     literal.
+   - **Usage:** Commonly used for string references without ownership or when dealing with parts of a string.
+   - **Example:**
+
+     ```rust
+     let string_literal: &str = "Hello, Rust!";
+     ```
+
+> **Note:** These string types cover various scenarios, from dynamic and mutable strings `String` to static and
+> immutable string slices `&str`. The type of string should be chosen depending on the specific requirements and the
+> characteristics of the data being manipulated.
+
+## Creating Strings
+
+1. **Using `String::new()`**:
+
+   ```rust
+   let mut empty_string = String::new();
+   ```
+
+   This creates a new, mutable, empty `String` that can be later modified and or assigned to a variable.
+
+2. **Using String Literals with `String::from`**:
+
+   ```rust
+   let greeting = String::from("Hello, World!");
+   ```
+
+   `String::from` allocates memory on the heap and initializes a new string with the specified string literal.
+
+## String Manipulation
+
+1. **Concatenation**
+
+   In Rust, there are multiple ways to concatenate strings depending on the given strings. Here is a list of
+   concatenation operators including examples:
+
+   > **`+` Operator:**
+
+   - **Usage:** concatenating two strings
+   - **Ownership:** takes ownership of the left operand
+   - **Borrowing:** To concatenate with a borrowed string (sliced out of another string using `&str`), it is necessary
+     to explicitly borrow it using the `&` operator.
+   - **Example:**
+
+     ```rust
+     let hello = String::from("Hello, ");
+     let world = String::from("World!");
+     let hello_world = hello + &world;  // here takes ownership of 'hello'
+     ```
```

Comment:
```suggestion
   - **Usage:** Concatenating two strings.
   - **Ownership:** Consumes the left operand.
   - **Borrowing:** To concatenate with a borrowed string (sliced out of another string using `&str`), it is necessary to explicitly borrow it using the `&` operator.
   - **Example:**

     ```rust
     let hello = String::from("Hello, ");
     let world = String::from("World!");
     let hello_world = hello + &world;  // Ownership of 'hello' is moved, 'world' is borrowed
     ```
```